### ISSUE TITLE
client/core: rethink scope of Login, expand API, separate concerns

### ISSUE THREAD

**Issue by chappjc** [decred/dcrdex#1762]
In the comments for the changes proposed in https://github.com/decred/dcrdex/pull/1761, I've described the heart of the issue as I see it.

The comment refers to the added mechanism to deal with concurrent Login requests at all:

```go
// The design issue that requires this is that we require multiple Login
// calls to succeed without a Logout between them. To properly solve this,
// Login would actually error if already logged in. However:
//  1. Login gives "restart" capabilities for loading and resuming trades.
//  2. We lack a mechanism for repeated callers (e.g. impatient HTTP
//     requests or sessions from other browsers) to get the result from the
//     already-running Login.
//
// Fundamentally, we need to separate the following notions:
//  1. login to Core (credentials check, get cookies)
//  2. Core startup (prep internal state, e.g. loading data from DB, DB upgrades, etc.)
//  3. login to configured servers (initializeDEXConnections > authDEX).
```

Purpose 1 would also do 2 if not already done.  If 2 already done, 1 would just check pw and get cookies for the other session.

A separate Core method may be needed to force re-running of 2 and/or 3.

Wallet connection could fall into either 2. or 3.

_Originally posted by @chappjc in https://github.com/decred/dcrdex/issues/1761#issuecomment-1205559884_

**Comment by ukane-philemon** [comment]
> 2. We lack a mechanism for repeated callers (e.g. impatient HTTP
//     requests or sessions from other browsers) to get the result from the
//     already-running Login

If we combine 2 (wallet connection & `resolveActiveTrades`) and 3 (`initializeDEXConnections`) into, say a `refresh` method. Can we do `refresh`  in a goroutine and return without waiting for it to finish? if not, what are the consequences?

**Comment by chappjc** [comment]
Something relevant and interesting: https://blog.devgenius.io/achieving-efficiency-with-singleflight-in-go-4daf668e48d
Not that we should take the plunge and use `singleflight`, but the article was interesting given this current issue.

**Comment by ukane-philemon** [comment]
> Can we do refresh in a goroutine and return without waiting for it to finish? if not, what are the consequences?

If `refresh` is run for each login in a goroutine, `Login` will return faster, and leave no room for impatient HTTP requests to trigger another goroutine. Unless we don't want to be on the markets page without finishing up with `refresh`?

**Comment by chappjc** [comment]
> If `refresh` is run for each login in a goroutine, `Login` will return faster

That's a possibility, but we have an existing contract with consumers of `Login` that a (non-nil err) return means ready to trade.

We can rethink anything about the API, but we have to basically make a flow chart of the different functionalities, how they should be accessed, etc.

**Comment by martonp** [comment]
So as I understand it, all the core startup/dex connections should be done only once on the first login, and afterwards, each login should just check credentials and return the proper information? And of course, concurrent requests should be handled properly.

I can work on this.
